db/inmemory.go

```diff
@@ -0,0 +1,186 @@
+package db
+
+import (
+	"errors"
+
+	"github.com/docker/go-plugins-helpers/volume"
+	"github.com/golang/glog"
+)
+
+// InMemoryVolumeDatabase defines a volume database that is completely in memory (and ephimeral)
+type InMemoryVolumeDatabase struct {
+	volumes map[string]*volume.Volume
+	mounts  map[string]map[string]int
+}
+
+// NewInMemoryVolumeDatabase creates a new InMemoryVolumeDatabase, inilizing all of its properties.
+func NewInMemoryVolumeDatabase() InMemoryVolumeDatabase {
+	warning := `
+********************************************************************************
+*                                                                              *
+*                                WARNING!!!                                    *
+*              DO NOT USE the "In-Memory Driver" IN PRODUCTION!                *
+*                                                                              *
+*        You are currently using the In-Memory Volume Database Driver,         *
+*        it is completely ephemeral and should only be used for testing        *
+*        purposes. When the plugin process stops, all data will be lost.       *
+*                                                                              *
+*        However, any data saved in the volumes will still be available        *
+*        by manually accessing the data in the storage backend filesystem.     *
+*                                                                              *
+********************************************************************************
+	`
+
+	glog.Warning(warning)
+
+	volumes := map[string]*volume.Volume{}
+	mounts := map[string]map[string]int{}
+	return InMemoryVolumeDatabase{volumes: volumes, mounts: mounts}
+}
+
+// Connect is a NOP, though required by VolumeDatabase interface
+func (i InMemoryVolumeDatabase) Connect() error {
+	glog.Info("Connect function called, no action taken.")
+	return nil
+}
+
+// Disconnect is a NOP, though required by VolumeDatabase interface
+func (i InMemoryVolumeDatabase) Disconnect() error {
+	glog.Info("Disconnect function called, no action taken.")
+	return nil
+}
+
+// Create a new volume in the database, returning an error if one occured.
+func (i InMemoryVolumeDatabase) Create(volumeName string, options map[string]string) error {
+	var exists bool
+	_, exists = i.volumes[volumeName]
+	if exists {
+		return errors.New("Volume already exists.")
+	}
+
+	_, exists = i.mounts[volumeName]
+	if exists {
+		return errors.New("Volume already exists.")
+	}
+
+	i.volumes[volumeName] = &volume.Volume{
+		Name:       volumeName,
+		Mountpoint: "",
+		Status:     nil}
+
+	return nil
+}
+
+// List all of the volumes in the database, returning an error if one occured.
+func (i InMemoryVolumeDatabase) List() ([]*volume.Volume, error) {
+	l := make([]*volume.Volume, 0, len(i.volumes))
```

Comment:
Short variable names hinder the reader's ability to understand this code. Please lengthen them for all our sakes.